app/src/editor/canvas/components/CodeEditor/index.jsx

```diff
@@ -1,101 +1,114 @@
+        setEditorLayout((layout) => ({
+            ...layout,
+            x,
+            y,
+        }))
+    }, [setEditorLayout])
+    const onEditorSizeUpdate = useCallback((width, height) => {
+        setEditorLayout((layout) => ({
+            ...layout,
+            width,
+            height,
+        }))
+    }, [setEditorLayout])
+    const onDebugPositionUpdate = useCallback((x, y) => {
+        setDebugLayout((layout) => ({
+            ...layout,
+            x,
+            y,
+        }))
+    }, [setDebugLayout])
+    const onDebugSizeUpdate = useCallback((width, height) => {
+        setDebugLayout((layout) => ({
+            ...layout,
+            width,
+            height,
+        }))
+    }, [setDebugLayout])
 
-    onShowEditor = () => {
-        this.setState({
-            editorOpen: true,
-        })
-    }
-
-    onCloseEditor = () => {
-        this.setState({
-            editorOpen: false,
-        })
-    }
-
-    onShowDebug = () => {
-        this.setState({
-            debugOpen: true,
-        })
-    }
-
-    onCloseDebug = () => {
-        this.setState({
-            debugOpen: false,
-        })
-    }
-
-    onEditorPositionUpdate = (x, y) => {
-        this.setState({
-            editorPosition: {
-                x,
-                y,
-            },
-        })
-    }
-
-    onDebugPositionUpdate = (x, y) => {
-        this.setState({
-            debugPosition: {
-                x,
-                y,
-            },
-        })
-    }
-
-    render() {
-        const {
-            readOnly,
-            code,
-            debugMessages,
-            onApply,
-            onChange,
-            onClearDebug,
-            children,
-        } = this.props
-        const { editorOpen, debugOpen, editorPosition, debugPosition } = this.state
-
-        return (
-            <React.Fragment>
-                {children && children(this.onShowEditor)}
-                {!!editorOpen && (
-                    <CodeEditorWindow
-                        position={editorPosition}
-                        onPositionUpdate={this.onEditorPositionUpdate}
-                        code={code}
-                        readOnly={readOnly}
-                        onClose={this.onCloseEditor}
-                        onApply={onApply}
-                        onChange={onChange}
-                        onShowDebug={this.onShowDebug}
-                    />
-                )}
-                {!!debugOpen && (
-                    <DebugWindow
-                        onPositionUpdate={this.onDebugPositionUpdate}
-                        position={debugPosition}
-                        messages={debugMessages}
-                        onClear={onClearDebug}
-                        onClose={this.onCloseDebug}
-                    />
-                )}
-            </React.Fragment>
-        )
-    }
+    return (
+        <React.Fragment>
+            {children && children(onShowEditor)}
```

Comment:
maybe `typeof children === 'function'` just so it doesn't totally blow up if children is an object or array?

Reply:
Enabling flow wouldn't prevent it but it could help. ;)